app/services/menu/item.rb

```diff
@@ -23,7 +23,7 @@ def initialize(name, options)
     end
 
     def url
-      add_relative_path(@url || @context.routes.url_for(url_hash.merge(:only_path=>true)))
+      add_relative_path(@url || @context.routes.url_for(url_hash.merge(:only_path=>true).except(:use_route)))
```

Comment:
This fixes the menu system (and its integration tests), as the url_for call here won't use `:use_route` so it was getting appended as a query parameter.